plugins/Diagnostics/Diagnostic/PrivateDirectories.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+use Piwik\Common;
+use Piwik\Filesystem;
+use Piwik\Http;
+use Piwik\SettingsPiwik;
+use Piwik\Translation\Translator;
+
+abstract class PrivateDirectories implements Diagnostic
+{
+    protected $privatePaths = [];
+
+    protected $labelKey = 'Diagnostics_RequiredPrivateDirectories';
+
+    /**
+     * @var Translator
+     */
+    protected $translator;
+
+    public function __construct(Translator $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function execute()
+    {
+        if (!SettingsPiwik::isMatomoInstalled()) {
+            return [];
+        }
+
+        $label = $this->translator->translate($this->labelKey);
+
+        $privatePaths = $this->privatePaths;
+
+        // create test file to check if tmp/empty exists Note: This won't work in a load balanced environment!
+        Filesystem::mkdir(PIWIK_INCLUDE_PATH . '/tmp');
+        file_put_contents(PIWIK_INCLUDE_PATH . '/tmp/empty', 'test');
```

Comment:
could we create this file in the `RecommendedPrivacyDirectory` check as it is only needed there?